cmd/ocm-backplane/managedJob/createManagedJob.go

```diff
@@ -335,3 +360,75 @@ func getJobStatus(client BackplaneApi.ClientInterface, job *BackplaneApi.Job) (B
 
 	return *formatJobResp.JSON200.JobStatus.Status, nil
 }
+
+var listDynaKube = func(cl client.Client,ctx context.Context, u client.ObjectList, opts ...client.ListOption) error{
+	return cl.List(ctx, u, opts...)
+}
+
+func getDynatraceInstance() (instance string, err error) {
+	bp, err := config.GetBackplaneConfiguration()
+	if err != nil {
+		return
+	}
+	cfg, err := login.GetRestConfig(bp, options.clusterID)
+	if err != nil {
+		return
+	}
+	cl, err := client.New(cfg, client.Options{})
+
+	if err != nil {
+		return
+	}
+	u := &unstructured.Unstructured{}
+	u.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "dynatrace.com",
+		Kind:    "DynaKube",
+		Version: "v1alpha1",
+	})
+
+	err = listDynaKube(cl,context.Background(), u, client.InNamespace("dynatrace"))
+	if err != nil {
+		return
+	}
+
+	err = u.EachListItem(func(object runtime.Object) error {
+		ux := object.(*unstructured.Unstructured)
+
+		data := ux.Object["spec"].(map[string]interface{})["apiUrl"].(string)
+		output := regexp.MustCompile(`https:\/\/(\w+)\..*/api`).FindStringSubmatch(data)
+		if len(output) > 1 {
+			instance = output[0]
+			//convert api url to ui
+			instance = strings.Replace(instance, ".live.", ".apps.", 1)
+			instance = strings.Replace(instance, "/api", "/ui", 1)
+		}
+		return nil
+	})
+
+	if err != nil{
+		return "", err
+	}
+	return
+}
+
+func isManagedCluster() (bool, error){
```

Comment:
Hey @tokalevasant , from our pairing session today, here's the new function we wrote that you can use in place of this:
```
func isManagedCluster() (bool, error) {
	//  and check for a label
	// with a key of ext-hypershift.openshift.io/cluster-type and a value of management-cluster
	connection, err := oocm.NewConnection().Build()
	if err != nil {
		return false, fmt.Errorf("failed to create OCM connection: %v", err)
	}
	defer connection.Close()

	labelsListResponse, err := connection.ClustersMgmt().V1().Clusters().Cluster(options.clusterID).ExternalConfiguration().Labels().List().Send()
	if err != nil {
		if labelsListResponse.Status() == 400 {
			return false, nil
		}
		return false, fmt.Errorf("failed to get label from ocm: %v", err)
	}

	for _, label := range labelsListResponse.Items().Slice() {
		if label.Key() == clusterTypeLabelKey {
			return label.Value() == clusterTypeLabelManagementCluster, nil
		}
	}

	return false, nil
}
```

Make sure you add the necessary constants as well to the top of the file:
```
const (
	clusterTypeLabelKey               = "ext-hypershift.openshift.io/cluster-type"
	clusterTypeLabelManagementCluster = "management-cluster"
)
```

Reply:
Changes completed, rebased and pushed